ziti/internal/edge_impl/listener.go

```diff
@@ -118,11 +143,47 @@ func NewMultiListener(serviceName string) MultiListener {
 			acceptC:     make(chan net.Conn),
 			errorC:      make(chan error),
 		},
+		listeners: map[edge.Listener]struct{}{},
 	}
 }
 
 type multiListener struct {
 	baseListener
+	listeners    map[edge.Listener]struct{}
+	listenerLock sync.Mutex
+}
+
+func (listener *multiListener) UpdateCost(cost uint16) error {
+	listener.listenerLock.Lock()
+	defer listener.listenerLock.Unlock()
+	for child := range listener.listeners {
+		if err := child.UpdateCost(cost); err != nil {
```

Comment:
if this fails mid-way through will some costs be updated and others not? should you try to update all of the listeners anyway and not fail-fast?

Reply:
I considered this, but decided to fast fail for expediency. I'll go back and take a look again to see how much more complicated this makes the code. Part of my reasoning for doing it this way is that this is likely happening as part of a graceful shutdown, and if it fails, we're still shutting down, just with slightly less grace. So it's more of a nicety. But on the other hand, it's always nice to maximize gracefulness.

Reply:
Updated to try and update all